### ISSUE TITLE
Vertical arfefacts in presentation software (specifically, in pdfpc)

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/cinnamon#10262]
```
 * Cinnamon version: 5.05
   - Please specify if you are using the daily builds PPA: no
 * Distribution - Mint 20.2
 * Graphics hardware *and* driver used: Intel; Mesa; external screen
 * 32 or 64 bit: 64.
 * I will provide logs if you need them.
 ```

**Issue**

As in [this closed beta issue](https://github.com/linuxmint/mint20.2-beta/issues/49?_pjax=%23repo-content-pjax-container), when I use [pdfpc](https://github.com/pdfpc/pdfpc) (latest version, compiled from git) on an 'extended' (as against 'mirrored' monitor), sometimes - not every time I run pdfpc - I get vertical artefacts of the type described in the beta issue. Here is a screenshot of my own:

![image](https://user-images.githubusercontent.com/47056543/128902515-2acff9d6-4cce-46f4-b6b2-ea83635c0509.png)

The bar on the right is the artefact (and I do not know where the letters in the artefact owe to).

**Steps to reproduce**

Install pdfpc.
Connect an external monitor to a laptop. Have Mint 'extend' the screen.
Run pdfpc: `pdfpc <name of pdf file, including .pdf extension>`.

**Expected behaviour**

No visual artefacts.

**Other information**

I am giving a presentation on Thursday afternoon!

Has the putative fix described in the beta issue been merged to mainline Mint? Or should I try the testing packages mentioned by that issue?

**Comment by mtwebster** [comment]
The fix has been merged but there's been no update yet. You can try the fix using packages from here:
https://github.com/linuxmint/muffin/releases/tag/master.mint20

Unpack the files (`tar xf packages.tar.gz`), enter the packages folder, and install using `sudo dpkg -i *.deb`. You may want to delete the -dbg and -doc packages first, as they would probably want to bring in additional dependencies you don't want or need.

If something goes wrong, the update manager would reinstall the current release packages.

**Comment by LinuxOnTheDesktop** [comment]
Thank you very much for the rapid, helpful response. I've installed those packages - excepting the two you mentioned - restarted Cinnamon and, so far, the presentation works.

**Comment by LinuxOnTheDesktop** [comment]
I note that I had to do the following to stop apt overwriting the packages that comprised the fix:

`sudo sudo apt-mark hold gir1.2-meta-muffin-0.0 libmuffin0 muffin-common muffin`

Also: in order to avoid

```
The following packages have unmet dependencies.
 muffin-dbg : Depends: libmuffin0 (= 5.0.1+uma) but 5.0.1 is installed
              Depends: muffin (= 5.0.1+uma) but 5.0.1 is installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

```

I had to do

`sudo apt remove muffin-dbg`

(even though I had not, to my knowledge, installed that package).

Do please issue the fix officially soon - people need to be able to use Mint to give presentations (and those who apply the fix could do without the apt hassles). Thank you.

**Comment by LinuxOnTheDesktop** [comment]
_Can I test the patch in some way for you?_ I wish to accelerate its deployment; using apt with 'held packages' is inconvenient. Thanks.

**Comment by LinuxOnTheDesktop** [comment]
Just now, and using the patched files, this happened:

`cinnamon[304942]: segfault at 18 ip 00007ff4f26a1bad sp 00007ffc98de8b28 error 4 in libmuffin.so.0.0.0[7ff4f265d000+74000]`

The circumstances are as follows.

Just now I received an update to Virtual Box (to version 6.1.26 r145957 - note though that for reason on my machine the full version number, as displayed within Virtual Box -> Help -> About is invisible but can be copied-and-pasted). After installing the update, I ran Virtual Box and launched my (longstanding, Windows) VM. Cinnamon crashed into fallback mode, but the VM ran. I restarted Cinnamon, exited the VM (and the VirtualBox Manager application) and then re-ran the VM. This time there was no crash.

I will try to find out, from elsewhere, whether users of the non-patched Muffin stuff see the same problem.

(As usual, MintReport shows no stack traces and, as I have said elsewhere, I do not even understand the MintReport interface.)
